src/main/java/io/github/ryotan/code/util/CodeEnumReflectionUtil.java

```diff
@@ -0,0 +1,290 @@
+        try {
+            Method method = code.getMethod(name);
+            if (isTarget(method, Filter.class, Predicate.class) && isValidCodeFilter(method.getGenericReturnType(), code)) {
+                return Optional.ofNullable((Predicate<C>) method.invoke(code));
+            }
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+            // nop
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * {@code type}が{@link ParameterizedType}のインスタンスで、型名が{@code code}と一致する場合{@code true}を返却します。
+     * 
+     * @param type 対象の型
+     * @param code 対象のクラス
+     * @return {@code type}の型名が{@code code}と一致する場合{@code true}
+     */
+    private static boolean isValidCodeFilter(Type type, Class<?> code) {
+        if (type instanceof ParameterizedType) {
+            final Type[] actualTypeArguments = ((ParameterizedType) type).getActualTypeArguments();
+            return actualTypeArguments.length == 1 && actualTypeArguments[0].getTypeName().equals(code.getTypeName());
+        }
+        return false;
+    }
+
+    /**
+     * {@code C}のコードのうち、{@code name}と一致する短縮論理名を取得します。
+     * 
+     * @param code 対象のコードのクラス
+     * @param name 短縮論理名
+     * @return 短縮論理名が示す値
+     * 
+     * @throws IllegalArgumentException {@code C}に短縮論理名が付与されている場合
+     */
+    public static <C extends CodeEnum<C>> String getShortLabelValue(C code, String name) {
+        try {
+            final Field field = code.getClass().getField(name);
+            final ShortLabel[] shortLabel = field.getAnnotationsByType(ShortLabel.class);
+            if (shortLabel.length != 0) {
+                return shortLabel[0].value();
+            }
+        } catch (NoSuchFieldException e) {
+            // nop
+        }
+        throw new IllegalArgumentException(String.format("%s.%s is not annotated with '@%s'.",
+                code.getClass().getSimpleName(), code, ShortLabel.class.getSimpleName()));
+    }
+
+    /**
+     * {@code code}クラスのうち、{@code maker}アノテーションが付与された{@code name}のフィールドもしくはメソッドの値を取得します。
+     * <p>
+     * {@code code}の{@code name}にアノテーションが見つからなかった場合は、{@link IllegalArgumentException}
+     * </p>
+     * 
+     * @param code   対象の{@code CodeEnum}クラス
+     * @param marker 対象のアノテーション
+     * @param name   対象のフィールド名もしくはメソッド名
+     * @return {@code code}の{@code name}に付与されているアノテーションの値
+     * 
+     * @throws IllegalArgumentException {@code code}の{@code name}にアノテーションが付与されていない場合
+     */
+    public static <C extends CodeEnum<C>> String getAnnotatedStringValue(C code, Class<? extends Annotation> marker, String name) {
+        return findAnnotatedStringValueFromField(code, marker, name)
+                .orElseGet(() -> findAnnotatedStringValueFromMethod(code, marker, name)
+                        .orElseThrow(() -> new IllegalArgumentException(String.format("The field or method annotated as '@%s' with name '%s'" +
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['+' should be on a new line.](https://www.codacy.com/app/ryotan/code-enum-support/file/2976023341/issues/source?bid=3375378&fileBranchId=3375797#l191)